core/src/main/java/org/apache/accumulo/core/metadata/schema/TabletMetadata.java

```diff
@@ -558,6 +559,26 @@ private static Optional<TServerInstance> checkServer(ClientContext context, Stri
         .map(address -> new TServerInstance(address, stat.getEphemeralOwner()));
   }
 
+  public static void validate(TabletMetadata tm) {
+    if (!tm.fetchedCols.contains(ColumnType.FILES) || !tm.sawPrevEndRow) {
```

Comment:
If we fetched files, may not see any files for a tablet can avoid some work by checking if the set is empty. The `sawPrevEndRow` is more for error checking.  If we fetched the PREV_ROW then its always expeced that it will be seen, so can check if it was fetched.  Other code will throw an error if prev row was fetched and not seen. 

```suggestion
    if (files.isEmpty() || !tm.fetchedCols.contains(ColumnType.PREV_ROW)) {
```

Reply:
When I was testing the code I noticed that when I made the change from `!tm.sawPrevEndRow` to `!tm.fetchedCols.contains(ColumnType.PREV_ROW)` it would throw an error for not having seen PrevEndRow. From my understanding fetchedCols should have PrevEndRow within it when the error happens but for some reason its not getting seen.

Reply:
Maybe part of the problem is that the test are using `new TabletMetadata.Builder();` which is a package private class that is only visible in the test because its in the same package.  Most test code uses `TabletMetadata.build(KeyExtent extent);` which is public and works very differently.   When using `new TabletMetadata.Builder();` its internal impl code that expected to be used in conjunction with `TabletMetadata.convertRow()`.  When using `TabletMetadata.build(KeyExtent extent);` it will call `TabletMetadata.convertRow()` which will call `new TabletMetadata.Builder();`.

This is very confusing, took a bit to figure out what was going on here.   Most of the time when writing test `TabletMetadata.Builder()` wil not be visible so there are not two confusing choices.  Because of the package this test is in, will see two different possible ways to do this.  Not a change for this PR, but we should probably rename `TabletMetadata.Builder` to something like `TabletMetadata.BuilderImpl`. 

For the new unit test in this PR it would probably be good to use `TabletMetadata.build(KeyExtent extent)` instead of ` TabletMetadata.Builder()`.  Doing that the code will get more realistic TabletMetadata objects that have gone through the same process that would happen when building a TabletMetadata object from Keys and Values.